### PR TITLE
dependency updates for last 3.7.x

### PR DESCRIPTION
**Pull request by Lex-2008**

### CODE REVIEW

**Comment by cf-bottom** [comment]
Thank you for submitting a pull request! Maybe @craigcomstock can review this?

**Comment by Lex-2008** [comment]
build: https://ci.cfengine.com/view/PR%20Pipeline/job/pr-pipeline/941/

**Comment by Lex-2008** [comment]
build: https://ci.cfengine.com/job/pr-pipeline/942/

**Comment by Lex-2008** [comment]
build: https://ci.cfengine.com/job/pr-pipeline/943/

**Comment by Lex-2008** [comment]
removed apache update - it required some fancier pcre-config. build: https://ci.cfengine.com/job/pr-pipeline/944/

File: deps-packaging/libyaml/hpux-header-fix.patch [hunk 1]
```diff
@@ -1,54 +1,32 @@
...
-  #define PTRDIFF_MAX _I64_MAX
-  #else
 *** a/src/Makefile.am	2015-04-22 14:10:00.418785980 +0200
 --- b/src/Makefile.am	2015-04-23 09:14:49.366662367 +0200
 ***************
 *** 1,4 ****
-! AM_CPPFLAGS = -I$(top_srcdir)/include
+! AM_CPPFLAGS = -I$(top_srcdir)/include -Wall
   lib_LTLIBRARIES = libyaml.la
   libyaml_la_SOURCES = yaml_private.h api.c reader.c scanner.c parser.c loader.c writer.c emitter.c dumper.c
```
Comment (craigcomstock) [inline deps-packaging/libyaml/hpux-header-fix.patch]: Why the addition of -Wall several places? Doesn't that just ask the compiler to be MORE picky? Basically a great idea but for a 3rd party lib I was just curious why it was added. :+1:

Comment (Lex-2008) [reply deps-packaging/libyaml/hpux-header-fix.patch]: it was added by upstream, so in order for these patch files to apply I had to add it to "delete" lines.
I decided to keep them in our "add" lines, too. In other words:

**old** patch file was like this:
```
- AM_CPPFLAGS = -I$(top_srcdir)/include
+ AM_CPPFLAGS = -I$(top_srcdir)/include -D_INCLUDE_STDC__SOURCE_199901 -D__STDC_CONSTANT_MACROS
```

**new** patch file is like this:
```
- AM_CPPFLAGS = -I$(top_srcdir)/include -Wall
+ AM_CPPFLAGS = -I$(top_srcdir)/include -D_INCLUDE_STDC__SOURCE_199901 -D__STDC_CONSTANT_MACROS -Wall
```

(hp-ux diff is in a bit different format, but the idea is like above)

Did it get clear now?

Comment (craigcomstock) [reply deps-packaging/libyaml/hpux-header-fix.patch]: That makes perfect sense. :seal-approves:

**Comment by Lex-2008** [comment]
3.7.x is not supported for quite some time already

**Pull request closed without merging**